actioncable/lib/action_cable/connection/base.rb

```diff
@@ -126,7 +126,8 @@ def on_message(message) # :nodoc:
       end
 
       def on_error(message) # :nodoc:
-        # ignore
+        # log errors to make diagnosing socket errors easier
+        logger.error "WebSocket error occurred: #{message}"
```

Comment:
In my case the problem only surfaced in production and not in development. Something like `undefined method write_nonblock for #<Puma::MiniSSL::Socket:0x007fe441e7ddd0>` should properly catch people's attention.